packages/gitgraph-react/src/Gitgraph.tsx

```diff
@@ -310,58 +307,152 @@ class Gitgraph extends React.Component<GitgraphProps, GitgraphState> {
     );
   }
 
-  private renderArrows(commit: Commit<ReactSvgElement>) {
-    const commitRadius = commit.style.dot.size;
+  private renderBranchLabel(commit: Commit<ReactSvgElement>) {
+    // gitgraph-core could compute branch labels into commits directly.
+    // That will make it easier to retrieve them, just like tags.
+    const branches = Array.from(this.gitgraph.branches.values());
+    return branches.map((branch) => {
+      if (!branch.style.label.display) return null;
 
-    return commit.parents.map((parentHash) => {
-      const parent = this.state.commits.find(({ hash }) => hash === parentHash);
-      if (!parent) return null;
+      const commitHash = this.gitgraph.refs.getCommit(branch.name);
+      if (commit.hash !== commitHash) return null;
 
-      // Starting point, relative to commit
-      const origin = this.gitgraph.reverseArrow
-        ? {
-            x: commitRadius + (parent.x - commit.x),
-            y: commitRadius + (parent.y - commit.y),
-          }
-        : { x: commitRadius, y: commitRadius };
+      // For the moment, we don't handle multiple branch labels.
+      // To do so, we'd need to reposition each of them appropriately.
```

Comment:
This shouldn't be too hard now. We can do just like tags.

Reply:
But we need to properly compute what branches should be displayed, per commit. For the moment, it's `branchToDisplay`, but it should be an array of the branches currently pointing to that commit.

The core could compute that.